llvm/lib/Transforms/AggressiveInstCombine/AggressiveInstCombine.cpp

```diff
@@ -1224,6 +1224,68 @@ static bool foldLibCalls(Instruction &I, TargetTransformInfo &TTI,
   return false;
 }
 
+static bool tryNarrowMathIfNoOverflow(Instruction &I, TargetTransformInfo &TTI,
+                                      const DataLayout &DL) {
+  unsigned opc = I.getOpcode();
+  Type *OldType = I.getType();
+  if (opc != Instruction::Add && opc != Instruction::Mul &&
+      !OldType->isIntOrIntVectorTy()) {
```

Comment:
Add/Mul only work with integer/intvector types.
```suggestion
      ) {
```